src/OpenTelemetry/Trace/Export/SimpleActivityProcessor.cs

```diff
@@ -41,6 +43,16 @@ public SimpleActivityProcessor(ActivityExporter exporter)
         /// <inheritdoc />
         public override void OnStart(Activity activity)
         {
+            // Perform sampling decision if source is empty.
+            // This occurs for activities created outside of ActivitySource
+            // i.e new Activity() directly.
+            // Note: This also can occur if one creates an ActivitySource of the name string.empty.
+            // In that case, the sampling logic would be run twice!
+            if (activity.Source.Name.Equals(string.Empty))
```

Comment:
this also looks hacky, as the sampling algorithm would be run more than once, if one creates ActivitySource with the name `string.Empty`.